tests/kolainst/destructive/auto-prune.sh

```diff
@@ -5,6 +5,47 @@ set -xeuo pipefail
 
 . ${KOLA_EXT_DATA}/libinsttest.sh
 
+journal_cursor() {
+    journalctl -o json -n 1 | jq -r '.["__CURSOR"]'
+}
+
+assert_journal_grep() {
+    local cursor re
+    cursor=$1
+    shift
+    re=$1
+    shift
+
+    if ! journalctl -u ostree-finalize-staged --after-cursor "${cursor}" --grep="$re" "$@" >/dev/null; then
+        fatal "failed to find in journal: $re"; exit 1
+    fi
+}
+
+assert_not_journal_grep() {
+    local cursor re
+    cursor=$1
+    shift
+    re=$1
+    shift
+
+    if journalctl -u ostree-finalize-staged --after-cursor "${cursor}" --grep="$re" "$@"; then
+        fatal "found in journal: $re"; exit 1
+    fi
+}
+
+inject_no_early_prune() {
+    mkdir -p /run/systemd/system/ostree-finalize-staged.service.d
+    cat >/run/systemd/system/ostree-finalize-staged.service.d/10-no-early-prune.conf << 'EOF'
+[Service]
+Environment=OSTREE_SYSROOT_OPTS=no-early-prune
+EOF
+    systemctl daemon-reload
+}
+
+remove_no_early_prune() {
+    rm -vf /run/systemd/system/ostree-finalize-staged.service.d/10-no-early-prune.conf
+}
```

Comment:
Meh, probably not worth the complexity from doing this from the service unit. We can just keep running `ostree admin finalize-staged` directly and the logs we need should still go to the journal.